lang/structs.go

```diff
@@ -311,10 +311,19 @@ func (obj *StmtRes) Init(data *interfaces.Data) error {
 	if err := obj.Name.Init(data); err != nil {
 		return err
 	}
+	typeMap := make(map[string]struct{})
 	for _, x := range obj.Contents {
 		if err := x.Init(data); err != nil {
 			return err
 		}
+
+		// checks if the current ResField type has already been added to typeMap
```

Comment:
Do you want this before or after the x.Init and why?

Reply:
Since you're asking, I'm guessing that you think it should be put before the x.Init(). I imagine we would do that to not waste computation on an object that shouldn't be initialised anyway due to the disallowance of duplicate fields in mcl.